src/ipc/StoreMap.cc

```diff
@@ -400,19 +405,15 @@ Ipc::StoreMap::openForReading(const cache_key *const key, sfileno &fileno)
     debugs(54, 5, "opening entry with key " << storeKeyText(key)
            << " for reading " << path);
     const int idx = fileNoByKey(key);
-    if (const Anchor *slot = openForReadingAt(idx)) {
-        if (slot->sameKey(key)) {
-            fileno = idx;
-            return slot; // locked for reading
-        }
-        slot->lock.unlockShared();
-        debugs(54, 7, "closed wrong-key entry " << idx << " for reading " << path);
+    if (const Anchor *slot = openForReadingAt(idx, key)) {
+        fileno = idx;
+        return slot; // locked for reading
```

Comment:
Since we are changing this code, let's polish it:

```suggestion
    if (const auto *anchor = openForReadingAt(idx, key)) {
        fileno = idx;
        return anchor; // locked for reading
```

Reply:
Done.